### ISSUE TITLE
Add linter that checks for manually included files being duplicate

### ISSUE THREAD

**Issue by BastianZim** [conda-forge/conda-smithy#1801]
I wanted to create an issue first to discuss if this makes sense because there are some downsides.

I wanted to add a function to the linter that does the following:

1. Check if there are license files listed in the meta.yaml
2. If yes, check if those files are included in the recipe directory
3. If they are, download the source and check if there are files with the same name
4. If there are, compare if the content is the same
5. If it is, post a linter hint that the manually included license file is a duplicate and that it is already available in the source, so it is not needed.

In text form: I sometimes see some PRs in staged-recipes where the license file is manually included but also available in the source, so it doesn't have to be included. This is a bit cumbersome to test because you need to download the source and manually compare. If we could have a linter function that does it, reviewing would be a bit faster.

Downside is primarily, that the linter would have to render the meta.yaml to replace jinja variables and download the source, which can take time. Additionally, the source can be big and clog up wherever the linter is running so we'd need checks for that as well. Finally, checking larger sources for files can also add time.

Thoughts?

**Comment by BastianZim** [comment]
@conda-forge/core for visibility

**Comment by carterbox** [comment]
This may be better implemented as a warning from conda-build if the LICENSE file squashes something from the source or LICENSE files from the source are unused.

**Comment by dopplershift** [comment]
I agree that conda-build is probably the right place, but could probably extend to any included file if the content is identical.

**Comment by BastianZim** [comment]
Conda-build makes sense but I think it should still be shown by the linter because not everyone goes through all of the logs. So feeding it back might make sense.

For all of the files: Would be nice but not sure how much time that adds?
